src/formatDate.js

```diff
@@ -9,6 +43,38 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+
+  const possibleDateParts = ['YYYY', 'YY', 'MM', 'DD'];
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
+
+  let datePartsMatched = {};
+
+  for (const datePart of possibleDateParts) {
+    const datePartIndex = fromFormat.indexOf(datePart);
+
+    if (datePartIndex !== -1) {
+      datePartsMatched[datePart] = oldDateParts[datePartIndex];
+    }
+  }
+  datePartsMatched = convertYear(datePartsMatched, fromFormat, toFormat);
+
+  const newDateParts = [];
+
+  for (const datePart of toFormat) {
+    if (!possibleDateParts.includes(datePart)) {
+      continue;
+    }
+
+    const newDatePart = datePartsMatched[datePart];
+
+    newDateParts.push(newDatePart);
+  }
```

Comment:
This loop is filtering out the separator from `toFormat`, but it's not immediately clear from the code. Consider adding a comment explaining that you're skipping the separator, or refactor the code to make it more explicit.